src/main/java/net/dries007/tfc/objects/te/TELargeVessel.java

```diff
@@ -0,0 +1,264 @@
+     * This method is only called server-side.
+     *
+     * @return The Packet that will be sent to clients in range.
+     */
+    @Override
+    @Nullable
+    public SPacketUpdateTileEntity getUpdatePacket()
+    {
+        updateLockStatus();
+        return super.getUpdatePacket();
+    }
+
+    /**
+     * Called on clients whenever this TileEntity received an update from the server.
+     **/
+    @Override
+    public void onDataPacket(NetworkManager net, SPacketUpdateTileEntity pkt)
+    {
+        readFromNBT(pkt.getNbtCompound());
+        updateLockStatus();
+    }
+
+    @Override
+    public boolean canInsert(int slot, ItemStack stack, EnumFacing side)
+    {
+        return !sealed && isItemValid(slot, stack);
+    }
+
+    @Override
+    public boolean canExtract(int slot, EnumFacing side)
+    {
+        return !sealed;
+    }
+
+    public void onSealed()
+    {
+        sealedTick = CalendarTFC.TOTAL_TIME.getTicks();
+        sealedCalendarTick = CalendarTFC.CALENDAR_TIME.getTicks();
+        TerraFirmaCraft.getNetwork().sendToDimension(new PacketLargeVesselUpdate(this, sealedCalendarTick), world.provider.getDimension());
+    }
+
+    public void onSeal()
+    {
+        for (int i = 0; i < 9; i++)
+        {
+            ItemStack stack = inventory.getStackInSlot(i);
+            if (!stack.isEmpty())
+            {
+                IFood cap = stack.getCapability(CapabilityFood.CAPABILITY, null);
+                if (cap != null)
+                {
+                    if (!cap.isRotten())
+                    {
+                        if (!cap.getTraits().contains(CapabilityFood.PRESERVED))
+                        {
+                            cap.getTraits().add(CapabilityFood.PRESERVED);
+                            cap.setCreationDate(CalendarTFC.PLAYER_TIME.getTicks() - (long)((CalendarTFC.PLAYER_TIME.getTicks() - cap.getCreationDate()) / CapabilityFood.PRESERVED.getDecayModifier()));
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    public void onUnseal()
+    {
+        for (int i = 0; i < 9; i++)
+        {
+            ItemStack stack = inventory.getStackInSlot(i);
+            if (!stack.isEmpty())
+            {
+                IFood cap = stack.getCapability(CapabilityFood.CAPABILITY, null);
+                if (cap != null)
+                {
+                    cap.getTraits().remove(CapabilityFood.PRESERVED);
+                    cap.setCreationDate(CalendarTFC.PLAYER_TIME.getTicks() - (long)((CalendarTFC.PLAYER_TIME.getTicks() - cap.getCreationDate()) * CapabilityFood.PRESERVED.getDecayModifier()));
+                }
+            }
+        }
+    }
+
+    public void onReceivePacket(long sealedCalendarTick)
+    {
+        this.sealedCalendarTick = sealedCalendarTick;
+    }
+
+    public boolean isSealed()
+    {
+        return sealed;
+    }
+
+    @Override
+    public void setAndUpdateSlots(int slot)
```

Comment:
Why is this overridden? (It skips a call to `markDirty()` when the inventory is updated.

Reply:
I didn't pay enought attention when i rewritten it to only have one container